tests/unit/services/cookies-test.js

```diff
@@ -332,72 +391,118 @@ describeModule('service:cookies', 'CookiesService', {}, function() {
 
       it('writes the value', function() {
         let value = randomString();
+        let subject = this.subject();
 
-        this.fakeFastBoot._fastBootInfo.response.cookie = function(name, theValue) {
-          expect(name).to.eq(COOKIE_NAME);
-          expect(theValue).to.eq(value);
+        subject._fastBoot.response.headers.append = function(...args) {
+          expect(args[0]).to.equal('cookie');
+          expect(args[1]).to.equal(`${COOKIE_NAME}=${value}`);
         };
 
-        this.subject().write(COOKIE_NAME, value);
+        subject.write(COOKIE_NAME, value);
+
+        let cookieValue = subject.get(`_fastBootCookiesCache.${COOKIE_NAME}.value`);
+        expect(cookieValue).to.equal(value);
       });
 
       it('URI-component-encodes the value', function() {
         let value = '!"§$%&/()=?"';
+        let encodedValue = encodeURIComponent(value);
+        let subject = this.subject();
 
-        this.fakeFastBoot._fastBootInfo.response.cookie = function(name, theValue) {
-          expect(theValue).to.eq(encodeURIComponent(value));
+        subject._fastBoot.response.headers.append = function(...args) {
+          expect(args[0]).to.equal('cookie');
+          expect(args[1]).to.equal(`${COOKIE_NAME}=${encodedValue}`);
         };
 
-        this.subject().write(COOKIE_NAME, value);
+        subject.write(COOKIE_NAME, value);
+
+        let cookieValue = subject.get(`_fastBootCookiesCache.${COOKIE_NAME}.value`);
+        expect(cookieValue).to.equal(encodedValue);
       });
 
       it('sets the cookie domain', function() {
-        this.fakeFastBoot._fastBootInfo.response.cookie = function(name, theValue, options) {
-          expect(options.domain).to.eq('example.com');
+        let domain = 'example.com';
+        let subject = this.subject();
+        this.fakeFastBoot.request.hostname = domain;
+
+        subject._fastBoot.response.headers.append = function(...args) {
+          expect(args[0]).to.equal('cookie');
+          expect(args[1]).to.equal(`${COOKIE_NAME}=test; domain=${domain}`);
         };
 
-        this.subject().write(COOKIE_NAME, 'test', { domain: 'example.com' });
+        subject.write(COOKIE_NAME, 'test', { domain });
+
+        let cookieDomain = subject.get(`_fastBootCookiesCache.${COOKIE_NAME}.options.domain`);
```

Comment:
I don't thing we should have tests asserting on the private caching API. The functionality is already tested in the read-after-write test anyway I think.